crypto/rand/randfile.c

```diff
@@ -280,8 +280,16 @@ const char *RAND_file_name(char *buf, size_t size)
     struct stat sb;
 #endif
 
+    /* Find the directory where the state file should go. */
+#ifdef OPENSSL_SYS_WINDOWS
+    if ((s = getenv("RANDFILE")) == NULL) {
+        s = getenv("TEMP");
+        if (s == NULL)
+            s = getenv("APPDATA");
```

Comment:
`APPDATA` is a *terrible* idea, exactly for the reason @mattcaswell states.  However, there are other environment variables to be used: `LOCALAPPDATA`, `TMP`

VC provides a function to get the location to use for temporary files as well, [have a look at how it does it](https://msdn.microsoft.com/en-us/library/windows/desktop/aa364992(v=vs.85).aspx) (also with env vars.